src/App/Application.cpp

```diff
@@ -2585,7 +2581,17 @@ void Application::initConfig(int argc, char ** argv)
     }
 
     variables_map vm;
-    parseProgramOptions(argc, argv, mConfig["ExeName"], vm);
+    {
+        BOOST_SCOPE_EXIT_ALL(&) {
+            // console-mode needs to be set (if possible) also in case parseProgramOptions
+            // throws, as it's needed when reporting such exceptions
+            if (vm.count("console")) {
+                mConfig["Console"] = "1";
+                mConfig["RunMode"] = "Cmd";
+            }

```

Comment:
I think I'm missing something, but why not just put this option before those that may throw exceptions?
Wouldn't that solve the issue too?

Reply:
The parsing code (`parseProgramOptions`), that already throws in case of `--help` only populates `vm`. The later code, that fills `mConfig` from `vm` (`processProgramOptions`) does not know the parser, thus it cannot generate the help text.